autometa/common/kmers.py

```diff
@@ -315,7 +317,8 @@ def count(
         logger.warning(f"counts already exist: {out} force to overwrite. [retrieving]")
         df = pd.read_csv(out, sep="\t", index_col="contig")
     else:
-        if not isinstance(size, int):
+        # checks if it is something like 12.0 vs. 12.9. Also check is an int
+        if not float(size).is_integer():
```

Comment:
We only care that size is an integer. We do not want to perform any conversion.

`if not isinstance(size, int)` seems to cover all of our bases here.